internal/storage/bucket_handle_test.go

```diff
@@ -417,6 +419,157 @@ func (testSuite *BucketHandleTest) TestCreateObjectMethodWhenGivenGenerationObje
 }
 
+func (testSuite *BucketHandleTest) TestBucketHandle_CreateObjectChunkWriter() {
+	var generation0 int64 = 0
+	var generationNon0 int64 = 786
+
+	tests := []struct {
+		name       string
+		generation *int64
+		expectErr  bool
+		objectName string
+		chunkSize  int
+	}{
+		{
+			name:       "NilGeneration",
+			generation: nil,
+			expectErr:  false,
+			objectName: "test_object_1",
+			chunkSize:  1024 * 1024,
+		},
+		{
+			name:       "GenerationAsZero",
+			generation: &generation0,
+			expectErr:  false,
+			objectName: "test_object_2",
+			chunkSize:  1024 * 1024,
+		},
+		{
+			name:       "NonZeroGeneration",
+			generation: &generationNon0,
+			expectErr:  true,
+			objectName: "",
+			chunkSize:  0,
+		},
+	}
+
+	for _, tt := range tests {
+		testSuite.T().Run(tt.name, func(t *testing.T) {
+			progressFunc := func(_ int64) {}
+			w, err := testSuite.bucketHandle.CreateObjectChunkWriter(context.Background(),
+				&gcs.CreateObjectRequest{
+					Name:                   tt.objectName,
+					GenerationPrecondition: tt.generation,
+				},
+				tt.chunkSize,
+				progressFunc,
+			)
+
+			if tt.expectErr {
+				assert.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				objWr, ok := (w).(*ObjectWriter)
+				require.True(t, ok)
+				require.NotNil(t, objWr)
+				assert.Equal(t, tt.objectName, objWr.ObjectName())
+				assert.Equal(t, tt.chunkSize, objWr.ChunkSize)
+				assert.Equal(t, reflect.ValueOf(progressFunc).Pointer(), reflect.ValueOf(objWr.ProgressFunc).Pointer())
+			}
+		})
+	}
+}
+
+func (testSuite *BucketHandleTest) TestBucketHandle_FinalizeUploadSuccess() {
+	var generation0 int64 = 0
+
+	tests := []struct {
+		name       string
+		generation *int64
+		objectName string
+		chunkSize  int
+	}{
+		{
+			name:       "NilGeneration",
+			generation: nil,
+			objectName: "test_object_1",
+			chunkSize:  1024 * 1024,
+		},
+		{
+			name:       "GenerationAsZero",
+			generation: &generation0,
+			objectName: "test_object_2",
+			chunkSize:  1024 * 100,
+		},
+	}
+
+	for _, tt := range tests {
+		testSuite.T().Run(tt.name, func(t *testing.T) {
+			progressFunc := func(_ int64) {}
+			wr, err := testSuite.bucketHandle.CreateObjectChunkWriter(context.Background(),
+				&gcs.CreateObjectRequest{
+					Name:                   tt.objectName,
+					GenerationPrecondition: tt.generation,
+				},
+				tt.chunkSize,
+				progressFunc,
+			)
+			require.NoError(t, err)
+			objWr, ok := (wr).(*ObjectWriter)
+			require.True(t, ok)
+			require.NotNil(t, objWr)
+			assert.Equal(t, tt.objectName, objWr.ObjectName())
+			assert.Equal(t, tt.chunkSize, objWr.ChunkSize)
+			assert.Equal(t, reflect.ValueOf(progressFunc).Pointer(), reflect.ValueOf(objWr.ProgressFunc).Pointer())
+
+			o, err := testSuite.bucketHandle.FinalizeUpload(context.Background(), wr)
+
+			assert.NoError(t, err)
+			assert.NotNil(testSuite.T(), o)
+		})
+	}
+}
+
+func (testSuite *BucketHandleTest) createObject(objName string) {
+	testSuite.T().Helper()
```

Comment:
I see this as definition for helper:
 Helper marks the calling function as a test helper function.
// When printing file and line information, that function will be skipped.
// Helper may be called simultaneously from multiple goroutines.

Don't we want to print file and line info when something goes wrong?

Reply:
This will still print the error but not with the file name and line number of test helper but with the calling Test function so it is easier to debug from where the error is coming from.

Eg: With Test Helper:
```
=== RUN   TestBucketHandleTestSuite/TestFinalizeUploadWithGenerationAsZeroWhenObjectAlreadyExists
    bucket_handle_test.go:554: 
        	Error Trace:	/usr/local/google/home/ashmeen/gcsfuse/internal/storage/bucket_handle_test.go:543
        	            				/usr/local/google/home/ashmeen/gcsfuse/internal/storage/bucket_handle_test.go:554
        	Error:      	An error is expected but got nil.
        	Test:       	TestBucketHandleTestSuite/TestFinalizeUploadWithGenerationAsZeroWhenObjectAlreadyExists
```

Without test helper:

```
=== RUN   TestBucketHandleTestSuite/TestFinalizeUploadWithGenerationAsZeroWhenObjectAlreadyExists
    bucket_handle_test.go:543: 
        	Error Trace:	/usr/local/google/home/ashmeen/gcsfuse/internal/storage/bucket_handle_test.go:543
        	            				/usr/local/google/home/ashmeen/gcsfuse/internal/storage/bucket_handle_test.go:554
        	Error:      	An error is expected but got nil.
        	Test:       	TestBucketHandleTestSuite/TestFinalizeUploadWithGenerationAsZeroWhenObjectAlreadyExists
--- FAIL: TestBucketHandleTestSuite (0.01s)
```